### PR TITLE
octopus: mgr/dashboard: Temporary User Lockout if 10 Invalid Login attempts

### PR DESCRIPTION
**Pull request by nizamial09**
backport tracker: https://tracker.ceph.com/issues/48794

backport of #38316
parent tracker: https://tracker.ceph.com/issues/40914
## Checklist
- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/dashboard/controllers/auth.py [hunk 1]
```diff
@@ -7,8 +7,10 @@
 from . import ApiController, RESTController, \
     allow_empty_body
 from .. import mgr
-from ..exceptions import DashboardException
+from ..exceptions import InvalidCredentialsError, UserDoesNotExist
 from ..services.auth import AuthManager, JwtManager
+from ..settings import Settings
+from . import ApiController, ControllerDoc, EndpointDoc, RESTController, allow_empty_body
```
Comment (aaSharma14) [inline src/pybind/mgr/dashboard/controllers/auth.py:13]:
```suggestion

```

Comment (aaSharma14) [reply src/pybind/mgr/dashboard/controllers/auth.py:13]: ControllerDoc, EndpointDoc - this needs to be removed as the Document REST API PR is not backported https://github.com/ceph/ceph/pull/36347

Comment (nizamial09) [reply src/pybind/mgr/dashboard/controllers/auth.py:13]: Thanks @aaSharma14 Will remove that.

**Comment by votdev** [comment]
Please run `tox` from `src/pybind/mgr/dashboard` before opening a PR, this will help you to identify issues like:
```
lint run-test: commands[0] | flake8
./controllers/auth.py:13:1: F811 redefinition of unused 'ApiController' from line 7
./controllers/auth.py:13:1: F811 redefinition of unused 'RESTController' from line 7
./controllers/auth.py:13:1: F811 redefinition of unused 'allow_empty_body' from line 7
./controllers/auth.py:13:1: F401 '.ControllerDoc' imported but unused
./controllers/auth.py:13:1: F401 '.EndpointDoc' imported but unused
2     F401 '.ControllerDoc' imported but unused
3     F811 redefinition of unused 'ApiController' from line 7
```

**Review by votdev — APPROVED** [review]
Successfully tested.

**Review by callithea — APPROVED** [review]
Lgtm

**Review by aaSharma14 — APPROVED** [review]

**Review by s0nea — APPROVED** [review]

**Pull request merged**